### PR TITLE
ENH: Improve Amulet support and documentation

### PR DESCRIPTION
**Pull request by ant0nsc**
- Move all Amulet-related code to a separate module
- Add bug fix for `RANK` environment to `hi-ml` runner
- Improve documentation and add standalone example script

### CODE REVIEW

File: docs/source/amulet/config.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  conda_yaml_file: $CONFIG_DIR/environment.yml
+
+code:
+  # local directory of the code. this will be uploaded to the server.
+  # $CONFIG_DIR is expanded to the directory of this config file
+  local_dir: $CONFIG_DIR/../../..
+
+storage:
+  data:
+    storage_account_name: innereye4data1
```
Comment (mebristo) [inline docs/source/amulet/config.yml:12]: Is it ok to have this storage account name checked in?

File: docs/source/amulet/config.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ sku: C1
+ command:
+ - python docs/source/amulet/amulet_script.py
+- name: Amulet Example with 2 GPUs on same node, don't delete RANK
+  sku: G2
+  command:
+  - python docs/source/amulet/amulet_script.py
+- name: Amulet Example with 2 nodes, 2 GPUs each
+  sku: 2xG2
+  # We only need to start the training script once per node. Pytorch Lightning
```
Comment (mebristo) [inline docs/source/amulet/config.yml:31]: This is a very useful comment - should we also highlight this in the documentation?

**Review by mebristo — APPROVED** [review]
LGTM!

File: hi-ml-cpath/src/health_cpath/configs/classification/BaseMIL.py [hunk 3]
```diff
@@ -73,6 +73,10 @@ class BaseMIL(LightningContainer, EncoderParams, PoolingParams):
                                                             "maximised (otherwise minimised).")
     ssl_checkpoint_run_id: str = param.String(default="", doc="Optional run id from which to load checkpoint if "
                                               "using SSLEncoder")
+    max_num_workers: int = param.Integer(10, bounds=(0, None),
```
Comment (kenza-bouzid) [inline hi-ml-cpath/src/health_cpath/configs/classification/BaseMIL.py:76]: 10 might be high as a default? what about 6? A lot of us faced some issues with num_workers and I think 6 was the limit that didn't give much headache.

**Review by kenza-bouzid — APPROVED** [review]
LGTM!

File: hi-ml-azure/src/health_azure/amulet.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        return True
+    return False
+
+
+def prepare_amulet_job() -> None:
+    """Make all necessary preparations to run the present training script as an Amulet job."""
+    # The RANK environment is set by Amulet, but not by AzureML. If set, PyTorch Lightning will think that all
+    # processes are running at rank 0 in its `rank_zero_only` decorator, which will cause the logging to fail.
+    if ENV_RANK in os.environ:
+        logging.info("Removing RANK environment variable.")
```
[the remark this answers is not in the document]
Comment (ant0nsc) [reply hi-ml-azure/src/health_azure/amulet.py:100]: I honestly don't know. The example and hi-ml are based on PL.

File: hi-ml-cpath/src/health_cpath/configs/classification/BaseMIL.py (hunk 3 above)
Comment (kenza-bouzid) [reply hi-ml-cpath/src/health_cpath/configs/classification/BaseMIL.py:76]: Cloud runs mostly, locally I can't go higher than 2

Comment (ant0nsc) [reply hi-ml-cpath/src/health_cpath/configs/classification/BaseMIL.py:76]: as discussed, will stick with 10 for now, as it has worked fine on the large VMs.

**Pull request merged**